stl/inc/cmath

```diff
@@ -1304,6 +1315,40 @@ template <class _Ty>
 _NODISCARD constexpr _Ty _Common_lerp(const _Ty _ArgA, const _Ty _ArgB, const _Ty _ArgT) noexcept {
     // on a line intersecting {(0.0, _ArgA), (1.0, _ArgB)}, return the Y value for X == _ArgT
 
+#ifdef _M_FP_STRICT
+    class _ExceptGuard {

```

Comment:
It's also allowed to change rounding mode under `/fp:precise` mode with `#pragma fenv_access(on)`.

Floating-point mode can be changed by [`#pragma float_control`](https://docs.microsoft.com/en-us/cpp/preprocessor/float-control?view=msvc-160), in which case `_M_FP_***` macros doesn't reflect the actual mode in effect.